src/plugins/intel_npu/src/CMakeLists.txt

```diff
@@ -17,6 +21,6 @@ if(ENABLE_ZEROAPI_BACKEND)
     add_subdirectory(backend)
 endif()
 
-if (NOT BUILD_COMPILER_FOR_DRIVER)
+if (ENABLE_NPU_PLUGIN_ENGINE)
```

Comment:
If plugin build is not supported when building CiD the original logic seems more clear/explicit to me. Not sure we need a new option.
We may want to discuss converging the build flows in longer term, so we would not need such exceptions.

Reply:
what do you mean by plugin build?

Reply:
I mean openvino_intel_npu_plugin target build

Reply:
I think the plugin should be built. We still need npu_al in that case